(PE-24670) Make status service respect tk-auth configs

amysahli opened the pull request:
Wrap status service routes so that access to the
status/v1/services endpoint can be configured to require
cert-authentication. If no authorization service is running,
the status service will default to allow unauthenticated
requests.

justinstoller commented:
We should do a quick update to the docs on how consumers of the status service can use/need to be aware of its interactions with tk-auth. Maybe an example auth.conf in [exposing-the-status-endpoint](https://github.com/puppetlabs/trapperkeeper-status/blob/master/documentation/developers.md#exposing-the-status-endpoint)?

Commit:
(PE-24670) Make status service respect tk-auth configs
Wrap status service routes so that access to the
status/v1/services endpoint can be configured to require
cert-authentication. If no authorization service is running,
the status service will default to allow unauthenticated
requests.

amysahli commented on documentation/developers.md:
```diff
@@ -78,6 +78,7 @@ webserver: {
     ssl-key: /etc/ssl/private_keys/myhostname.pem
     ssl-ca-cert: /etc/ssl/certs/ca.pem
     default-server: true
+    client-auth: "want"
```
@justinstoller is this needed?

justinstoller replied:
This was required for your test that created an https request but didn't provide it's own cert. (we wanted to verify it got a 403 rather than https error). I don't believe it's required for this documentation.

amysahli commented on documentation/developers.md:
```diff
@@ -88,10 +89,29 @@ web-router-service: {
 
+The status service can interact with the tk-auth service if it is also a
+dependency. If it is a dependency, it is possible to require authentication for
+the status endpoint.  You can do this by updating the puppetserver/auth.conf file
```
^^ is puppetserver/auth.conf correct?

justinstoller commented on documentation/developers.md:
```diff
@@ -88,10 +89,29 @@ web-router-service: {
 protected server, see the [Status Proxy documentation](./status-proxy-service.md).
 
+The status service can interact with the tk-auth service if it is also a
+dependency. If it is a dependency, it is possible to require authentication for
```
It'd be good to distinguish between jar/library dependencies (which tk-auth is now one of for tk-status) vs services loaded during application runtime. Maybe something like, "if it's included in the trapperkeeper application's bootstrap config" rather than "dependency".

amysahli replied:
It's not included in the trapperkeeper application's bootstrap config, is it?  I thought it was in the application's (like puppetserver or puppetdb) bootstrap config file.

justinstoller replied:
I don't think that's a distinction I meant to make, I used "trapperkeeper application" as a generalization of puppetdb, puppetserver, etc. ...Applications using trapperkeeper.

justinstoller commented on documentation/developers.md:
```diff
@@ -88,10 +89,29 @@ web-router-service: {
+The status service can interact with the tk-auth service if it is also a
+dependency. If it is a dependency, it is possible to require authentication for
+the status endpoint.  You can do this by updating the
+/etc/puppetlabs/puppetserver/conf.d/auth.conf file to include this authorization rule:
```
I'd just say something like "tk-auth's configuration file" or something less specific than the actual path to where Puppet Server puts its tk-auth file (assuming the primary consumers of this documentation are either people outside the Puppet Server team within Puppet or community members using tk-status as a library in their own applications).

justinstoller commented on test/puppetlabs/trapperkeeper/services/status/status_service_test.clj:
```diff
@@ -192,6 +207,61 @@
+                                   :name "puppetlabs status service"}]}})
+        (testing "without authentication returns 403"
+          (let [resp (http-client/get "https://localhost:8180/status/v1/services" incomplete-ssl-config)]
+            (is (= 403 (:status resp)))))
```
This causes an error message logged during the test run. Can we wrap this with a `with-test-logging` block to keep that from happening?
